### PR TITLE
feat: added filters on balance page

### PR DESCRIPTION
**Pull request by CzarekDryl**
https://tw.auroracreation.com/app/projects/562144/tasks/15577548

Filtering table on Balance page:
- "All filter button"
- Show pills for active filters 
- Allow user to filter by Tokens
- Allow user to filter by Attributes: 
	- Native  
	- Reputation (it's hidden, as it's not supported yet)

Search for tokens by name & symbol:
- As soon as a user starts typing in the search input, the ‘X’ icon shows.
- When selected, the X icon clears the input entered by the user and reseats the dynamic search results in the  table.
- This feature should be implemented across all dropdown menus that have a search including for filter menus.
- Empty state should be displayed if no tokens found.

Empty states:
- Display No Results component if there are no results of the search and filter

## Description

- Added filters to balance page

## Out of scope

- Sorting

### CODE REVIEW

**Review by rdig — APPROVED** [review]
Can't find much to point at, so nice  work. :+1: I left a comment that I want you to address before merging though

Screenshot from 2024-02-29 22-23-46

**Review by warrencolliver — APPROVED** [review]
looks good to me, happy to re-approve if needed after Raul's comments

File: src/components/frame/v5/pages/BalancePage/partials/BalanceTable/Filters/partials/filters/AttributeFilters/consts.ts
Comment (CzarekDryl) [reply src/components/frame/v5/pages/BalancePage/partials/BalanceTable/Filters/partials/filters/AttributeFilters/consts.ts:14]: Yup, I've added TODO for this

**Review by iamsamgibbs — COMMENTED** [review]
Great work on this issue, all looks great. I just wanted to confirm this is intended for the filtering, the 'Native' attribute in effect overrides the token type? With these filters selected I would have expected no results:

Screenshot 2024-03-01 at 10 49 31

**Review by rdig — CHANGES_REQUESTED** [review]
Good pickup @iamsamgibbs. I totally missed that.

You are correct in expecting having that table empty with those filter selected. Filter order / hierarchy is something that definitely needs to be addressed

**Review by iamsamgibbs — APPROVED** [review]
Thanks for the fix, all working as expected! 🎉

**Review by warrencolliver — DISMISSED** [review]

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @CzarekDryl! Nice work on this one.

Two minor change requests.

1. The input in the active state has funky border radius which has been an issue before - 

image


2. On mobile, the search title in the search pop up should be 'SEARCH BALANCE ACTIVITY'

Thank you!

**Review by rdig — APPROVED** [review]
This works as expected now. Nice work! :100: 

Screenshot from 2024-03-05 21-39-32

**Review by iamsamgibbs — APPROVED** [review]

**Review by melyndav — APPROVED** [review]
Thanks for the changes @CzarekDryl. Approved from my side.

**Pull request merged**
